tests/binary-expressions/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -208,13 +253,48 @@ prevState = prevState ||
     selectedCatalog: null
   };
 
+prevState =
+  prevState ||
```

Comment:
Why does this one break the `prevState` onto the next line, while the one on line 266 doesn't?

Reply:
@suchipi As I understand it, the object is a subexpression of the right side group. So `shouldInlineLogicalExpression` is false for the outermost group, but true for the inner group in the right side.

See [playground](https://prettier.io/playground/#%7B%22content%22%3A%22const%20a%20%3D%20longlonglonglonglonglonglong%20%7C%7C%20longlonglonglonglonglonglong%3B%5Cn%5Cnconst%20a%20%3D%20longlonglonglonglonglonglong%20%7C%7C%20longlonglonglonglonglonglong%20%7C%7C%20%7B%5Cn%20%20prop%5Cn%7D%22%2C%22options%22%3A%7B%22ast%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%2C%22jsxBracketSameLine%22%3Afalse%2C%22output2%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22printWidth%22%3A50%2C%22semi%22%3Atrue%2C%22singleQuote%22%3Afalse%2C%22tabWidth%22%3A2%2C%22trailingComma%22%3A%22none%22%2C%22useTabs%22%3Afalse%7D%7D)